components/camel-telegram/src/main/java/org/apache/camel/component/telegram/TelegramEndpoint.java

```diff
@@ -61,6 +61,12 @@ public Exchange createExchange(Update update) {
             if (update.getMessage().getChat() != null) {
                 exchange.getIn().setHeader(TelegramConstants.TELEGRAM_CHAT_ID, update.getMessage().getChat().getId());
             }
+        }else if(update.getChannelPost() != null) {
```

Comment:
Still style issues, please run mvn with -Psoursecheck option and squash commits after fix and before push (use -f git option) them again.